raiwidgets/raiwidgets/fairness_metric_calculation.py

```diff
@@ -13,6 +19,98 @@
     "needs to provide a MetricFrame class to calculate metrics. For an " \
     "refer to fairlearn.metrics.MetricFrame"
 
+z_score = 1.959964  # This z-score corresponds to the 95% confidence interval
+alpha = 0.95        # Conventional level of power for statistical tests
+digits_of_precision = 4
+
+
+def compute_wilson_bounds(p, n, digits=digits_of_precision, z=z_score):
+    """ Returns lower and upper bound (Binomial Proportion)
+    """
+    denominator = 1 + z**2 / n
+    centre_adjusted_probability = p + z * z / (2 * n)
+    adjusted_standard_deviation = np.sqrt(
+        (p * (1 - p) + z * z / (4 * n))) / np.sqrt(n)
+    lower_bound = (centre_adjusted_probability - z *
+                   adjusted_standard_deviation) / denominator
+    upper_bound = (centre_adjusted_probability + z *
+                   adjusted_standard_deviation) / denominator
+    return (round(lower_bound, digits), round(upper_bound, digits))
+
+
+# Unused
+def compute_standard_normal_error_binomial(metric_value, sample_size, z_score):
+    """ Standard Error Calculation (Binary Classification)
+
+    Assumes infinitely large population,
+    Should be used when the sampling fraction is small.
+    For sampling fraction > 5%, may want to use finite population correction
+    https://en.wikipedia.org/wiki/Margin_of_error
+
+    Note:
+        Returns absolute error (%)
+    """
+    return z_score * np.sqrt(metric_value * (1.0 - metric_value)) \
+        / np.sqrt(sample_size)
+
+
+# def compute_standard_normal_error(metric_value, y_true, y_pred,\
+#                                   sample_size, z_score):
+#     """ Standard Error Calculation
+
+#     Attempting to generalize for regression/probability tasks. Incomplete
+
+#     """
+
+#     # return z_score * np.sqrt(VARIANCE HERE) / np.sqrt(sample_size)
+#     return
+
+
+def wilson_wrapper(y_true, y_pred, func):
+    assert len(y_true) == len(y_pred)
+    p = func(y_true, y_pred)
+    n = len(y_true)
+    result = compute_wilson_bounds(p, n, digits_of_precision, z_score)
+    if float('nan') in result:
+        return [None, None]
+    return result
+
+# custom recall/precision error bar functions to have n =/= len(y_pred)
+# because it should be n = (tp + fn) and n = (tp+fp), respectively
+
+
+def recall_wilson(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    tn, fp, fn, tp = confusion_matrix(y_true, y_pred).ravel()
+    bounds = compute_wilson_bounds(
+        tp / (tp + fn), tp + fn, digits_of_precision, z_score)
+    return bounds
+
+
+def precision_wilson(y_true, y_pred):
```

Comment:
For testing things like this, at a minimum please create a couple of short (say 10 element) input arrays, and have a test that the output value is unchanged. It's not a comprehensive test, but it's better than nothing.